qiskit/visualization/pulse/matplotlib.py

```diff
@@ -32,7 +32,11 @@
                                    SnapshotChannel, Channel)
 from qiskit.pulse import (Waveform, Snapshot, Play,
                           Acquire, PulseError, ParametricPulse, SetFrequency, ShiftPhase,
-                          Instruction, ScheduleComponent, ShiftFrequency, SetPhase)
+                          Instruction, Schedule, ShiftFrequency, SetPhase)
+
+
+ScheduleComponent = Union[Schedule, Instruction]
+"""An element that composes a pulse schedule."""
```

Comment:
Any suggestion of the place? I think defining this at schedule.py will cause cyclic import. So having this definition in __init__ of the top module is reasonable, but I'm not sure if you prefer this. Otherwise I will create types.py which has no dependency as I did for the visualization modules.